test/e2e/helm_test.go

```diff
@@ -184,14 +191,14 @@ var _ = Describe("Create a proper set of manifests when using helm charts", func
 
 	It("should deploy all providers with latest version", func() {
 		manifests, err := helmChart.Run(map[string]string{
-			"configSecret.name":      "test-secret-name",
-			"configSecret.namespace": "test-secret-namespace",
-			"core":                   "cluster-api",
-			"controlPlane":           "kubeadm",
-			"bootstrap":              "kubeadm",
-			"infrastructure":         "docker",
-			"ipam":                   "in-cluster",
-			"addon":                  "helm",
+			"configSecret.name":          "test-secret-name",
+			"configSecret.namespace":     "test-secret-namespace",
+			"core.cluster-api.name":      "null",
```

Comment:
The `name` field doesn't actually exist and is never used for anything. This is just here to make the types line up, if anyone has a better solution I'm all ears.

Reply:
Yes, I am not sure if that make sense, this is just to set a key. that's why I was suggesting also adding `enabled`. This would solve problem with empty dictionaries values

```suggestion
			"core.cluster-api.enabled":      "true",
```
 looking here https://helm.sh/docs/intro/using_helm/#the-format-and-limitations-of---set

you might be able to do

```suggestion
			"core":      "cluster-api",
```
or 
```suggestion
			"core.cluster-api":      "null",
```

I do like `enabled` more since it allow to easily turn off a provider.

Reply:
Just pushed the `enabled` version.